### DISCUSSION TITLE
Nothing to see here.

### DISCUSSION THREAD

**Discussion by vdegenne** [ibhagwan/fzf-lua#1543] [Q&A] [answered]
Is there a way to tell fzf-lua to keep executing the origin call of a picker? I am not sure how to explain so here's what I try to do:

I made this action for when a file is selected:

```lua
config.defaults.actions.files["default"] = function(selected, opts)
  local entry = path.entry_to_file(selected[1], opts, opts._uri)

  open_or_focus(entry.path)
end
```

it will focus the selected file or open it if not found in the windows list. It worked well until I tried to use 

```lua
require("fzf-lua").lsp_document_symbols({
  regex_filter = symbols_filter
})
```
I understand now the symbols picker is calling `actions.files["default"]` action as well, in my case my function will be called and will fail. I could use the uri and line and column information to do everything myself, but I don't want to reinvent the wheel since `lsp_document_symbols` (and `lsp_live_workspace_symbols`!) is supposed to do that for me.

It would be good to have a way to tell lsp pickers apart from real file picker.

Or at least a way to tell fzf-lua to disregard a particular execution, e.g.

```lua
config.defaults.actions.files["default"] = function(selected, opts)
  local entry = path.entry_to_file(selected[1], opts, opts._uri)

  if opts.__INFO.cmd == "lsp_document_symbols" then
    return false -- Nothing to see here
  end

  open_or_focus(entry.path)
end
```

Ngl, I'm kinda stuck here and don't know how to fix that.

**Comment by ibhagwan** [comment, marked as answer]
Maybe I’m missing something but why don’t you use `actions.file_switch_or_edit`?

https://github.com/ibhagwan/fzf-lua/blob/175ddbb9bbe28eef006516f38b126215de1fef57/lua/fzf-lua/actions.lua#L371

**Reply by vdegenne** [reply]
omg!

**Reply by vdegenne** [reply]
@ibhagwan Is there a similar action but to switch or (v/h)split? I am using a very long vertical screen (with maximize window feature) and usually I don't want to open a file "on top" of another because I assume if a window is open it has a meaning (and a special place) and it should stay visible unless intentionally destroyed(!), so I don't mind having a lot of windows opened. I want to *see* my cursor switch to the appropriate window or *see* a new window being created with the new buffer. Thanks for your help.

**Reply by vdegenne** [reply]
ok looking at the source code that looks easy to implement:

```lua
M.file_switch_or_vsplit = function(selected, opts)
  if not M.file_switch({ selected[1] }, opts) then
    M.file_vsplit({ selected[1] }, opts)
  end
end
```

Do you think that's something you could add to the plugin (`file_switch_or_split` and `file_switch_or_vsplit`)?

If you want I could PR that change.

**Reply by ibhagwan** [reply]
> Do you think that's something you could add to the plugin (file_switch_or_split and file_switch_or_vsplit)?

Given this is so easy to add/customize I really don’t think it’s needed, why don’t you add these few lined to your config instead?

**Reply by vdegenne** [reply]
Sure I don't mind, it was just to make these directly available to other users.

**Reply by ibhagwan** [reply]
> Sure I don't mind, it was just to make these directly available to other users.

I understand, on the other hand then why not also have `file_switch_or_tab` also, etc? You see where this is going.

**Reply by vdegenne** [reply]
Yes right! TBH I'm not even sure if I need that anyways, sometimes simpler things are the best. Thanks again
